packages/connect-wallet-modal/src/components/WalletsModal/components/ConnectWalletModal/ConnectWalletModal.tsx

```diff
@@ -52,27 +52,26 @@ export const ConnectWalletModal = ({
 
   const isWalletsToggleButtonShown = useRef(false);
 
-  const [walletsList, setWalletsList] = useState<WalletConnectorData[]>([]);
-  const [walletsListFull, setWalletsListFull] = useState<WalletConnectorData[]>(
+  const [walletsList, setWalletsList] = useState<WalletAdapterData[]>([]);
+  const [walletsListFull, setWalletsListFull] = useState<WalletAdapterData[]>(
     [],
   );
 
-  const getWalletsListFull = useCallback(async () => {
-    // Asynchronously filling wallets list because there is an async wallet detection during wallets sorting.
-    // Actually, almost all wallets can be detected synchronously, so this process is expected to be fast,
-    // and a loading indicator is not required for now.
-    const _walletsListFull = await sortWalletsList({
-      config,
-      walletConnectorsList,
-      walletsShown,
-      walletsPinned,
-    });
-    setWalletsListFull(_walletsListFull);
-  }, [config, walletConnectorsList, walletsPinned, walletsShown]);
-
   useEffect(() => {
-    void getWalletsListFull();
-  }, [getWalletsListFull]);
+    const fetch = async () => {
+      // Asynchronously filling wallets list because there is an async wallet detection during wallets sorting.
+      // Actually, almost all wallets can be detected synchronously, so this process is expected to be fast,
+      // and a loading indicator is not required for now.
+      const _walletsListFull = await sortWalletsList({
+        config,
+        walletConnectorsList,
+        walletsShown,
+        walletsPinned,
+      });
+      setWalletsListFull(_walletsListFull);
+    };
+    void fetch();
+  }, [config, walletConnectorsList, walletsPinned, walletsShown]);
```

Comment:
This is prone to race condition, if deps change for any reason while function is running, you could have two racing setStates. 
Let's at least add isLoading state, it can act as simple guard to this effect(`if(isLoading) return`).  Latter can used for ui if needed.